core/sodasql/dataset_analyzer.py

```diff
@@ -66,7 +66,7 @@ def analyze(self, warehouse: Warehouse, table_name: str):
                     qualified_regex = dialect.qualify_regex(format_regex)
                     regexp_like = dialect.sql_expr_regexp_like(
                         qualified_column_name, qualified_regex)
-                    count_field = f'COUNT(CASE WHEN {regexp_like} THEN 1 END)'
+                    count_field = f'COUNT(CASE WHEN {regexp_like} THEN 1 END) AS {validity_format}'
```

Comment:
An encoding error was thrown some times. As [this issues](https://github.com/mkleehammer/pyodbc/issues/328) explained, it could happen due to special characters in the column name. The regexes contain special characters and without an alias they are present in the column name. so this problem was resolved by adding an alias (which do not contain special characters).